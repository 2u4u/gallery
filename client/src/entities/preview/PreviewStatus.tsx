import { PreviewType } from 'shared/types/index.type';

type Props = {
    preview: PreviewType;
}

function PreviewStatus({ preview }: Props) {
    return (
        <div className='preview__status'>
            {
                preview.downloaded ? 
                    <img 
                        src={`./img/${preview.downloaded}.svg`} 
                        alt={preview.downloaded} 
                        className={preview.downloaded === 'loading' ? 'preview__spin' : 'preview__svg'}
                    /> : 
                    null
            }
        </div>
    );
}

export default PreviewStatus;
