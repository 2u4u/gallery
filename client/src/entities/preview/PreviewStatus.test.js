import renderer from 'react-test-renderer';
import PreviewStatus from './PreviewStatus';

describe('Preview status component ', () => {
    const preview = {
        type: 'image',
        url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
        name: 'test image',
        file: 'Fake file',
    };
    it('matches snapshot', () => {
        const component = renderer.create(
            <PreviewStatus preview={preview} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
