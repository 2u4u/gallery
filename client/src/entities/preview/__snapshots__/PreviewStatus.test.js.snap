// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview status component  matches snapshot 1`] = `
<div
  className="preview__status"
/>
`;
